src/creatures/creature.cpp

```diff
@@ -1592,6 +1592,17 @@ LightInfo Creature::getCreatureLight() const {
 	return internalLight;
 }
 
+uint16_t Creature::getSpeed() const {
+	const auto speed = baseSpeed + varSpeed;
+	if (speed >= 0 && speed <= std::numeric_limits<uint16_t>::max()) {
+		return speed;
+	} else if (speed < 0) {
+		return 0;
+	} else {
+		return std::numeric_limits<uint16_t>::max();
+	}
+}
```

Comment:
You can change by std::clamp(value, min, max)

```c++
uint16_t Creature::getSpeed() const {
     return static_cast<uint16_t>(std::clamp<int>(baseSpeed + varSpeed, 0, std::numeric_limits<uint16_t>::max()));
}
```

Reply:
You are right, I totally forgot about this function. Fixed!